warrior/Actions/CliActions/cli_actions.py

```diff
@@ -820,10 +820,60 @@ def send_testdata_command_kw(self, system_name, session_name=None, wdesc='', var
                                                                   session_name=session_name,
                                                                   datafile=self.datafile)
 
-        td_resp_dict = get_object_from_datarepository(str(session_id)+"_td_response")
-        if not WarriorCliClass.cmdprint:
-            td_resp_dict.update(resp_dict)
-
+        temp_testdata_dict = data_Utils.get_command_details_from_testdata(testdata, varconfigfile,
```

Comment:
Can you move this part of code to respective util methods to avoid placing computation logic in keywords.
This will also help you in accessing available variables and reusing local methods.